src/task/tenth/MyHashMap.java

```diff
@@ -0,0 +1,118 @@
+package task.tenth;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class MyHashMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private Node<K, V>[] entries = new Node[DEFAULT_INITIAL_CAPACITY];
+
+
+    public void put(K key, V value) {
+        if (checkSize()) {
+            grow();
+        }
+        int index = indexBacket(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        if (entries[index] == null) {
+            entries[index] = newNode;
+        } else {
+            Node<K, V> entry = entries[index];
+            if ((entry.getKey() == null && key == null) || entry.getKey().equals(key)) {
+                entry.setValue(value);
+                return;
+            }
+            entry = entry.next;
+            while ((entry) != null) {
+                if (key.equals(entry.next.getKey())) {
+                    entry.next.setValue(value);
+                    return;
+                }
+                entry = entry.next;
+            }
+            entry = newNode;
+        }
+        size++;
+    }
+
+    public V get(K key) {
+        int index = indexBacket(key);
+        Node<K, V> node = entries[index];
+        while (node != null) {
+            if ((key == null && index == 0) | (key.equals(node.getKey()))) {
+                return node.value;
+            }
+            node = node.next;
+        }
+        return null;
+    }
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private void grow() {
+        Node<K, V>[] temp = (Node<K, V>[]) new Node[2 * entries.length];
+        for (int i = 0; i < entries.length; i++) {
+            Node<K, V> node = entries[i];
+            while (node != null) {
+                int newIndex = indexBacket(node.key);
+                if (temp[newIndex] == null) {
+                    temp[newIndex] = node;
+                    node = node.next;
+                } else {
+                    Node<K, V> nodeSecond = temp[newIndex];
+                    while (nodeSecond != null) {
+                        if (nodeSecond.next == null) {
+                            nodeSecond.next = node;
```

Comment:
we need add checking key.equals(entry.next.getKey()) here

Reply:
Sorry, I don't understand. It is metod grow, and it overwrite old values with new index.  It can't repeat keys, because I set new value in metod put, when keys are equals.  line24-26